File: app/src/main/java/com/unis/gameplatfrom/adapter/MainAdapter.java
package com.unis.gameplatfrom.adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.util.Log;
import android.view.View;
import android.widget.ImageView;
import android.widget.ProgressBar;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.chad.library.adapter.base.BaseViewHolder;
import com.squareup.picasso.Picasso;
import com.unis.gameplatfrom.R;
import com.unis.gameplatfrom.manager.GlideManager;
import com.unis.gameplatfrom.entity.GamesEntity;

import java.util.List;

public class MainAdapter extends BaseEmptyViewAdapter<GamesEntity>  {


    private int number = 1;
    private List<GamesEntity> list;
    private Context mContext;

    public MainAdapter(Context mContext, List<GamesEntity> list) {
        super(mContext,true,R.layout.item_main, list);
        this.mContext = mContext;
        this.list = list;
    }


    @Override
    public void onBindViewHolder(@NonNull BaseViewHolder holder, int position, @NonNull List<Object> payloads) {


        Log.e("xxxxx","xxxxxx"+payloads.size());
        if(payloads.isEmpty()){
            super.onBindViewHolder(holder, position);

        }else {



        }

//        int number = (int) payloads.get(0);
//
//            if(position == number){
//
//
//
//        }

//        GamesEntity gamesEntity = list.get(position);
//
//        ProgressBar progressBar = holder.getView(R.id.item_progress);
//        progressBar.setProgress(gamesEntity.getProgress());

        //TextView






    }

    @Override
    protected void convert(BaseViewHolder holder, GamesEntity entity) {

//        GlideManager.loadImg(entity.getIcon(), holder.getView(R.id.item_games_icon));
//        StringBuilder builder = new StringBuilder();
//        builder.append(entity.getName());
//        holder.setText(R.id.item_games_name, builder.reverse().toString()+"：名戏游");
//
//
//        //holder.setText(R.id.item_games_version, "版本："+entity.getV());
//
//
//
//        if(entity.isNewGame()){
//            holder.setText(R.id.item_games_version, (entity.getV()-1)+":本版");
//            holder.setVisible(R.id.item_games_new_version,true);
//            holder.setText(R.id.item_games_new_version, entity.getV()+":本版新现发");
//
//        }else {
//            holder.setText(R.id.item_games_version, entity.getV()+":本版");
//            holder.setVisible(R.id.item_games_new_version,false);
//        }
//
//
//        if(entity.isGame()){
//            holder.setVisible(R.id.item_games_down,true);
//        }else {
//            holder.setVisible(R.id.item_games_down,false);
//        }


        RelativeLayout relativeLayout = holder.getView(R.id.game_item_layout);

        relativeLayout.getBackground().setAlpha(50);

        ImageView imageView = holder.getView(R.id.item_games_icon);

       // GlideManager.loadImg(entity.getIcon(), holder.getView(R.id.item_games_icon));
        Picasso.with(mContext)
                .load(entity.getIcon())
                .placeholder(R.drawable.logo)
                .into(imageView);

        holder.setText(R.id.item_games_name, "游戏名："+entity.getName());


        //holder.setText(R.id.item_games_version, "版本："+entity.getV());

        TextView down = holder.getView(R.id.item_games_down);
        TextView install = holder.getView(R.id.item_games_install);
        TextView newVersion = holder.getView(R.id.item_games_new_version);
        holder.setText(R.id.game_number,String.valueOf(holder.getAdapterPosition()+1));


        if(entity.isNewGame() && entity.getNewVersion() != 0 ){
            holder.setText(R.id.item_games_version, "版本："+(entity.getNewVersion()));
            newVersion.setVisibility(View.VISIBLE);
            holder.setText(R.id.item_games_new_version, "发现新版本："+entity.getV());

        }else {
            holder.setText(R.id.item_games_version, "版本："+entity.getV());

            newVersion.setVisibility(View.INVISIBLE);
        }



        if(entity.isDownGame()){

            holder.setVisible(R.id.lin_progress,true);

        }else {

            holder.setVisible(R.id.lin_progress,false);
        }


        holder.setText(R.id.item_progress_number,String.valueOf(entity.getProgress())+"%");

        ProgressBar progressBar = holder.getView(R.id.item_progress);
        progressBar.setProgress(entity.getProgress());


        if(entity.isInstallGame() && entity.isLocal()){
            install.setVisibility(View.VISIBLE);
            down.setVisibility(View.INVISIBLE);

        }else if(!entity.isInstallGame() && entity.isLocal()){
            down.setVisibility(View.VISIBLE);
            install.setVisibility(View.INVISIBLE);
        }else {
            down.setVisibility(View.INVISIBLE);
            install.setVisibility(View.INVISIBLE);
        }
    }



}

File: kotlin/src/main/java/com/unis/kotlin/api/adapter/MainAdapter.kt
package com.unis.kotlin.api.adapter

import android.content.Context
import android.view.View
import android.widget.ImageView
import android.widget.ProgressBar
import android.widget.RelativeLayout
import android.widget.TextView
import com.chad.library.adapter.base.BaseViewHolder
import com.squareup.picasso.Picasso
import com.unis.kotlin.R
import com.unis.kotlin.entity.GamesEntity

class MainAdapter :BaseEmptyViewAdapter<GamesEntity>{

    private var context : Context? = null

    constructor(context: Context,list: ArrayList<GamesEntity>) :
            super(context, R.layout.item_games,list){
        this.context = context
    }


    override fun convert(holder: BaseViewHolder?, entity: GamesEntity?) {

        var relativeLayout =  holder!!.getView<RelativeLayout>(R.id.game_item_layout)
        relativeLayout.background.alpha = 50

        var imageView = holder.getView<ImageView>(R.id.item_games_icon)

        Picasso.with(mContext).load(entity!!.icon).placeholder(R.drawable.logo).into(imageView)

        holder.setText(R.id.item_games_name, "游戏名：" + entity!!.name)

        val down = holder.getView<TextView>(R.id.item_games_down)
        val install = holder.getView<TextView>(R.id.item_games_install)
        val newVersion = holder.getView<TextView>(R.id.item_games_new_version)


        holder.setText(R.id.game_number, "" + (holder.adapterPosition + 1).toString())

        if (entity.isNewGame && entity.newVersion != 0) {
            holder.setText(R.id.item_games_version, "版本：" + entity.newVersion)
            newVersion.visibility = View.VISIBLE
            holder.setText(R.id.item_games_new_version, "发现新版本：" + entity.v)

        } else {
            holder.setText(R.id.item_games_version, "版本：" + entity.v)

            newVersion.visibility = View.INVISIBLE
        }



        if (entity.isDownGame) {

            holder.setVisible(R.id.lin_progress, true)

        } else {

            holder.setVisible(R.id.lin_progress, false)
        }


        holder.setText(R.id.item_progress_number, entity.process.toString() + "%")


        //holder.setText(R.id.item_progress_kb,
        // speed <= 0 ? "" : String.format("%s/s", CommonUtil.formatFileSize(speed)))


        val progressBar = holder.getView<ProgressBar>(R.id.item_progress)
        progressBar.progress = entity.process


        if (entity.isInstallGame && entity.isLocal) {
            install.visibility = View.VISIBLE
            down.visibility = View.INVISIBLE

        } else if (!entity.isInstallGame && entity.isLocal) {

            down.visibility = View.VISIBLE
            install.visibility = View.INVISIBLE

        } else {
            down.visibility = View.INVISIBLE
            install.visibility = View.INVISIBLE
        }


        holder.addOnClickListener(R.id.item_game_down)
        holder.addOnClickListener(R.id.item_game_cancel)




    }
}